### PR TITLE
fix shouldHideEvent saying an event is a leave/join when a profile ch…

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/riot-web/issues/6231

### CODE REVIEW

File: src/shouldHideEvent.js [hunk 1]
```diff
@@ -27,10 +27,11 @@ function memberEventDiff(ev) {
     const content = ev.getContent();
     const prevContent = ev.getPrevContent();
 
-    diff.isJoin = content.membership === 'join' && prevContent.membership !== 'ban';
-    diff.isPart = content.membership === 'leave' && ev.getStateKey() === ev.getSender();
-
     const isJoinToJoin = content.membership === prevContent.membership && content.membership === 'join';
+
+    diff.isJoin = !isJoinToJoin && content.membership === 'join' && prevContent.membership !== 'ban';
+    diff.isPart = !isJoinToJoin && content.membership === 'leave' && ev.getStateKey() === ev.getSender();
```
Comment (lukebarnard1) [inline src/shouldHideEvent.js]: But a "part" could never be a join to join.. so `!isJoinToJoin` is always true here.

Comment (turt2live) [reply src/shouldHideEvent.js]:
```
const isMembershipChanged = content.membership !== prevContent.membership;
diff.isJoin = isMembershipChanged && ...;
diff.isPart = isMembershipChanged && ...;
```

is what I had before @t3chguy beat me to the PR.

**Comment by t3chguy** [comment]
@turt2live be my guest, I'm about to board a flight to won't be able to fix this PR for a few hours

**Comment by turt2live** [comment]
I already deleted the branch, and you have a PR open, and it's not that major of an issue to me.

**Comment by lukebarnard1** [comment]
Who's got the conn. on this then?

**Comment by t3chguy** [comment]
@lukebarnard1 sorry for delays, turns out I had 0% of the time I envisioned in Switzerland

**Pull request merged**